### PR TITLE
Tweaks to cmake build

### PR DESCRIPTION
**Pull request by wavexx**
Giving this a go and tweaking a few things along the way.

Being able to build more than a single variant in the same tree is.. intriguing. Didn't think of doing it this way.

I would have some further proposals:

- move utils/bootstrap.py to tools/, since we already have it. Or do the opposite (so that we have one generic "tooldir").
- strip -E3Dv6full and 1_75_mm from the variant file names directly (that is, by renaming them). 

It's ok to strip them like we do, but I want to avoid dumb filename conflicts if you just *happen* to have a second pre-stripped filename in there.

@3d-gussner would renaming 1_75mm_MK3S-EINSy10a-E3Dv6full.h to MK3S-EINSy10a be an issue?

### CODE REVIEW

**Comment by 3d-gussner** [comment]
@wavexx I think the `1_75mm_...-E3Dv6full` is for the past as there have been [2.85mm and v5 hotends](https://help.prusa3d.com/guide/introduction_132507).
I think we can rename them as they are only used in the source code and the build scripts. 

- [ ] Update `build.sh`
- [ ] Update `PF-build.sh`
- [ ] Update `.travis.yml`
- [ ] Update `build server scripts`
- [ ] Update `README.md`

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# do not build the firmware by default (tests are the focus if not crosscompiling)
+project(cmake_test)
+
+# include catch_discover_tests function from Catch2
+include(${Catch2_SOURCE_DIR}/contrib/Catch.cmake)
+
+# Make test executable
+set(TEST_SOURCES
```
Comment (3d-gussner) [inline tests/CMakeLists.txt:8]: These have been removed from `.travis.yml`

**Comment by wavexx** [comment]
Does the build server rely on the name somehow?

**Comment by 3d-gussner** [comment]
The build server uses the different variants files and these are "hard-coded". But easy to change.

**Comment by wavexx** [comment]
Didn't check the sizes/languages yet, but it looks like that now, _finally_, the FW itself is reproducible, which I consider a major step forward.

Doesn't set a fancy date yet, but we'll get that from the last git commit somehow.

**Comment by wavexx** [comment]
I'm also getting perfectly matching binary sizes now on the EN-only build.

**Comment by wavexx** [comment]
Going to stop here for this PR ;)

File: CMakeLists.txt [hunk 2]
```diff
@@ -60,6 +60,7 @@ SET(SELECTED_LANGUAGES ${MAIN_LANGUAGES} ${COMMUNITY_LANGUAGES})
 
 get_dependency_directory(prusa3dboards PRUSA_BOARDS_DIR)
 project(Prusa-Firmware)
+add_subdirectory(lib)
```
Comment (vintagepc) [inline CMakeLists.txt:63]: This should probably be excluded if the compiler is avr.

File: CMakeLists.txt [hunk 3]
```diff
@@ -173,12 +184,6 @@ SET(FW_SOURCES
 )
 list(TRANSFORM FW_SOURCES PREPEND ${CMAKE_CURRENT_SOURCE_DIR}/Firmware/)
 
-foreach(_FILE ${FW_SOURCES})
-    get_filename_component(_BASE ${_FILE} NAME)
-    set_property(SOURCE ${_FILE} APPEND_STRING PROPERTY COMPILE_FLAGS "-frandom-seed=${_BASE}.o")
-endforeach()
-
-
```
Comment (vintagepc) [inline CMakeLists.txt:181]: Why did you remove this? Now your compiler will use random seeds all the time...

File: CMakeLists.txt [hunk 4]
```diff
@@ -458,42 +477,29 @@ endfunction()
 
 if(CMAKE_CROSSCOMPILING)
 
-	file(GLOB FW_VARIANTS RELATIVE ${PROJECT_SOURCE_DIR}/Firmware/variants ${PROJECT_SOURCE_DIR}/Firmware/variants/*.h)
-	foreach(THIS_VAR IN LISTS FW_VARIANTS)
-		string(REPLACE ".h" "" TRIMMED_NAME "${THIS_VAR}")
-		message("Variant added: ${TRIMMED_NAME}")
-        string(REPLACE "-E3Dv6full" "" DIR_NAME "${TRIMMED_NAME}")
```
Comment (vintagepc) [inline CMakeLists.txt:465]: this was done so as not to break the existing builds... I'd leave it until we actually make the change, the prefix and postfix are just garbage/noise on screen and add to directory lengths/paths to anannoying degree.

Comment (vintagepc) [reply CMakeLists.txt:465]: nevermind - stupid diff view is terrible sometimes.

**Comment by wavexx** [comment]
Closing
